test/integration/022_bigquery_test/test_bigquery_adapter_specific.py

```diff
@@ -0,0 +1,37 @@
+""""Test adapter specific config options."""
+from test.integration.base import DBTIntegrationTest, use_profile
+import textwrap
+import yaml
+
+
+class TestBigqueryAdapterSpecific(DBTIntegrationTest):
+
+    @property
+    def schema(self):
+        return "bigquery_test_022"
+
+    @property
+    def models(self):
+        return "adapter-specific-models"
+
+    @property
+    def profile_config(self):
+        return self.bigquery_profile()
+
+    @property
+    def project_config(self):
+        return yaml.safe_load(textwrap.dedent('''\
+        config-version: 2
+        models:
+            test:
+                materialized: table
+                expiring_table:
+                    time_to_expiration: 4    
+        '''))
+
+    @use_profile('bigquery')
+    def test_bigquery_time_to_expiration(self):
+        _, stdout = self.run_dbt_and_capture()
+        self.assertIn(
+            'expiration_timestamp: TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL '
+            '4 hour)', stdout)
```

Comment:
I think the example I copied here was one that expected failure on the model, so the query would be dumped in `stdout`.

I probably want to inspect `results` from `self.run_dbt()`, but could use a pointer to the compiled SQL within the results to do this `assertIn` (it's a little hard to decipher the schema sometimes). Let me know if that makes sense

Reply:
I think you want `results[index].node.injected_sql`. You can look for results by node name using `results[index].node.name`.

Also, don't feel at all obligated to do this, but because we use pytest for tests now you are free to use the (much more ergonomic, at least to me) `assert whatever in stdout` syntax.

Reply:
@beckjake The error I got makes me think `injected_sql` isn't what I'm looking for in results.

`E       AssertionError: 'expiration_timestamp: TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL 4 hour)' not found in 'select 1 as id'`

This config adds the expiration_timestamp as part of the ddl, and if this was ddl, it should say something like `create or replace table as ...`. I can't remember if this is present in debug (which I believe just dumps the query), or where else the full ddl might be in the results. Any ideas?

Reply:
Ah, now that I look at this more carefully, I think this will be in the output if you run with `--debug`, but not the `injected_sql`. `injected_sql` contains the value that will end up as the `sql` value in the materialization. But this change happens ultimately in the `create_table_as` macro that's called from the materialization.

It is, I suppose, always possible that we don't log all our queries on bigquery? That would be pretty bad behavior on our part.

Reply:
I ran this on a real project locally and it doesn't look like the ddl is anywhere in run_results.json, but it definitely is in the output with `--debug`.